pkg/asset/installconfig/openstack/validation/platform.go

```diff
@@ -38,7 +38,7 @@ func validateMachinesSubnet(p *openstack.Platform, n *types.Networking, ci *Clou
 		if len(p.ExternalDNS) > 0 {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("externalDNS"), p.ExternalDNS, "externalDNS is set, externalDNS is not supported when machinesSubnet is set"))
 		}
-		if !validUUIDv4(p.MachinesSubnet) {
+		if !validUUIDv4(p.MachinesSubnet) || ci.MachinesSubnet == nil {
```

Comment:
We should rather use the `field.NotFound` error for consistency:
```suggestion
		if ci.MachinesSubnet == nil {
			allErrs = append(allErrs, field.NotFound(fldPath.Child("machinesSubnet"), p.MachinesSubnet))
		}
		if !validUUIDv4(p.MachinesSubnet) {
```